src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccountDecider.fs

```diff
@@ -0,0 +1,71 @@
+﻿module Deciders.BankAccountDecider
+
+open System
+open BankAccount
+
+type OpenBankAccount =
+    { BankAccountId: Guid
+      AccountNumber: string
+      ClientId: Guid
+      CurrencyISOCode: string }
+
+type RecordDeposit = { Amount: decimal; CashierId: Guid }
+type WithdrawCashFromATM = { Amount: decimal; AtmId: Guid }
+type CloseBankAccount = { Reason: string }
+
+type Command =
+    | OpenBankAccount of OpenBankAccount
+    | RecordDeposit of RecordDeposit
+    | WithdrawnCashFromATM of WithdrawCashFromATM
+    | CloseBankAccount of CloseBankAccount
+
+let openBankAccount now command =
+    { BankAccountId = command.BankAccountId
+      AccountNumber = command.AccountNumber
+      ClientId = command.ClientId
+      CurrencyISOCode = command.CurrencyISOCode
+      CreatedAt = now ()
+      Version = 1 }
+    |> BankAccountOpened
+
+let recordDeposit now (command: RecordDeposit, bankAccount) =
+    if (bankAccount.Status = BankAccountStatus.Closed) then
+        invalidOp "Account is closed!"
+
+    { BankAccountId = bankAccount.Id
+      Amount = command.Amount
+      CashierId = command.CashierId
+      RecordedAt = now ()
+      Version = bankAccount.Version + 1L }
+    |> DepositRecorded
+
+let withdrawCashFromATM now (command, bankAccount) =
+    if (bankAccount.Status = BankAccountStatus.Closed) then
+        invalidOp "Account is closed!"
+
+    if (bankAccount.Balance < command.Amount) then
+        invalidOp "Account is closed!"
```

Comment:
Is 'Account is closed!' the correct response?

Reply:
Nope, it wasn't here's the correct one: https://github.com/oskardudycz/Ogooreck/pull/13/commits/9a3f781d7183a75d8b313714a8e239f2dc18dfe5